addons/website_slides/static/src/js/slides_course_slides_list.js

```diff
@@ -52,12 +54,19 @@ publicWidget.registry.websiteSlidesCourseSlidesList = publicWidget.Widget.extend
      * @private
      */
     _checkForEmptySections: function (){
-        this.$('.o_wslides_js_slides_list_container ul').each(function (){
-            var $emptyCategory = $(this).find('.o_wslides_js_slides_list_empty');
-            if ($(this).find('li.o_wslides_slides_list_slide[data-slide-id]').length === 0) {
-                $emptyCategory.removeClass('d-none').addClass('d-flex');
-            } else {
-                $emptyCategory.addClass('d-none').removeClass('d-flex');
+        var self = this;
+        this.$('.o_wslides_slide_list_category > ul').each(function (){
+            var categoryID = $(this).data('categoryId');
+            var categorySlideCount = $(this).find('.o_wslides_slides_list_slide:not(.o_not_editable)').length;
+            var $emptyFlagContainer = self.$('.o_wslides_slide_list_category_header[data-category-id="'+ categoryID +'"] > div:first-child');
+            var $emptyFlag = $emptyFlagContainer.find('small');
+            if (categorySlideCount === 0 && $emptyFlag.length === 0){
+                $emptyFlagContainer.append($('<small>', {
+                    'class': "ml-1 text-muted",
+                    text: _t("Empty")
+                }));
+            } else if (categorySlideCount > 0 && $emptyFlag.length > 0){
+                $emptyFlag.remove();
             }
```

Comment:
After some testing in my console:

- Loop on `o_wslides_slide_list_category` instead.
  That way you can do `$(this).find('.o_wslides_slide_list_category_header')` directly, no need to do the categoryId trick.
- Not sure about the ` > div:first-child` part, can you remove it or find something more simple?
- I would not try to find the container and then test for a 'small' tag.
  Instead, add a specific class on your small flag (`o_wslides_no_category_flag`) and test if it's there with `$(this).find('o_wslides_no_category_flag').length`
  Then you only need to find the appropriate container when you want to append.
  So `$(this).find('o_wslides_slide_list_category_header [appropriate selector?]').append($('<small/>', { ... })`

All this to try to simplify the code a little.
A find the new version very "code heavy" compared to the previous one.